### PR TITLE
Throw exceptions on incorrect serialization inputs

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by certik** [comment]
The change seems fine. The CI failure seems to be caused by this PR.

File: symengine/tests/basic/test_serialize-cereal.cpp [hunk 1]
```diff
@@ -73,3 +76,16 @@ TEST_CASE("Test serialization using cereal", "[serialize-cereal]")
         real_mpfr(mpfr_class("0.35", 100, 10)));
 #endif
 }
+
+TEST_CASE("Test serialization exception", "[serialize-cereal]")
+{
+    RCP<const Basic> expr = se::parse("x + y");
+    std::string orig_data = expr->dumps();
+    std::vector<int> positions = {4, 8, 9, 15, 16};
```
Comment (bjodah) [inline symengine/tests/basic/test_serialize-cereal.cpp:84]: Can we have a comment here on how these positions were chosen? Random, or carefully calculated?

**Comment by bjodah** [comment]
Looks good!

**Comment by bjodah** [comment]
Wonderful!

**Pull request merged**